src/org/zaproxy/zap/extension/ascanrulesAlpha/TestUserAgent.java

```diff
@@ -0,0 +1,187 @@
+    private static final String INTERNET_EXPLORER_7 = "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)";
+    private static final String INTERNET_EXPLORER_6 = "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)";
+    private static final String GOOGLE_BOT_2_1 = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)";
+    private static final String MSN_BOT_1_1 = "msnbot/1.1 (+http://search.msn.com/msnbot.htm)";
+    private static final String YAHOO_SLURP = "Mozilla/5.0 (compatible; Yahoo! Slurp; http://help.yahoo.com/help/us/ysearch/slurp)";
+    private static final String I_PHONE_3 = "Mozilla/5.0 (iPhone; U; CPU iPhone OS 3_0 like Mac OS X; en-us) AppleWebKit/528.18 (KHTML, like Gecko) Version/4.0 Mobile/7A341 Safari/528.16";
+
+    private static final String[] USER_AGENTS = {
+            INTERNET_EXPLORER_8,
+            INTERNET_EXPLORER_7,
+            INTERNET_EXPLORER_6,
+            GOOGLE_BOT_2_1,
+            MSN_BOT_1_1,
+            YAHOO_SLURP,
+            I_PHONE_3
+    };
+
+    private HttpMessage originalMsg;
+    private int originalResponseBodyHash;
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String[] getDependency() {
+        return null;
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INFO_GATHER;
+    }
+
+    @Override
+    public String getSolution() {
+        return "";
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public void scan() {
+        setOriginalMessageAndResponseBodyHash();
+        if(originalMsg == null){
+            log.warn("Aborting due to empty original message leads to false positives");
+            return;
+        }
+
+        for(String userAgent: USER_AGENTS){
+            if (isStop()) {
+                return;
+            }
+            attack(userAgent);
+        }
+    }
+
+    private void setOriginalMessageAndResponseBodyHash() {
+        originalMsg = getOriginalMessage();
+        if(originalMsg != null){
+            originalResponseBodyHash = originalMsg.getResponseBody().hashCode();
+        }
+    }
+
+    private HttpMessage getOriginalMessage(){
+        HttpMessage baseMsg = getBaseMsg();
+        if(baseMsg.getResponseHeader().isEmpty()){
+            return resendBaseMessage();
+        }
+        return baseMsg;
+    }
+
+    private HttpMessage resendBaseMessage(){
+        try {
+            HttpMessage newMsg = getNewMsg();
+            sendAndReceive(newMsg);
+            if(!newMsg.getResponseHeader().isEmpty()){
```

Comment:
This check is not necessary, either the message is sent/received or an exception thrown.